admin/code/LeftAndMain.php

```diff
@@ -1260,14 +1353,27 @@ public function getEditForm($id = null, $fields = null) {
 			$actionsFlattened = $actions->dataFields();
 			if($actionsFlattened) foreach($actionsFlattened as $action) $action->setUseButtonTag(true);
 
-			$form = CMSForm::create(
+			$negotiator = $this->getResponseNegotiator();
+			$form = Form::create(
 				$this, "EditForm", $fields, $actions
 			)->setHTMLID('Form_EditForm');
-			$form->setResponseNegotiator($this->getResponseNegotiator());
 			$form->addExtraClass('cms-edit-form');
 			$form->loadDataFrom($record);
 			$form->setTemplate($this->getTemplatesWithSuffix('_EditForm'));
 			$form->setAttribute('data-pjax-fragment', 'CurrentForm');
+			$form->setValidationResponseCallback(function() use ($negotiator, $form) {
```

Comment:
`SecurityAdmin` uses `parent::getEditForm()` which gets the callback from `LeftAndMain`.
`ModelAdmin` uses `GridFieldDetailForm`, which in turn still uses its own PJAX logic (in `generateValidationResponse()`), I've confirmed a ModelAdmin can still display validation errors.
Once these interfaces are switched over to React, requiring JSON responses, we can look into migrating the new logic there.